### PR TITLE
Adds toy Gloves of the North Star

### PR DESCRIPTION
**Pull request by JosephJomama**
## About The Pull Request

Adds toy Gloves of the North Star, called "Gloves of the South Star", to the arcade machine. These work exactly the same as Gloves of the North Star, except your punches are also made incapable of doing any damage. They are also incompatible with any martial arts, so no worries about the Sleeping Carp guy getting his hands in these and killing the entire station. 

## Why It's Good For The Game

Adds another fun, cool prize to the arcade machine! Prank your friends! Prank the HoS! What's not to love?
(objectively this is pretty stupid but I thought it was a funny idea and would be a cool addition to the fake traitor item prize pool collection so I spent like 2 days coding it help)

## Changelog
:cl:
add: New from Donk. Co: Gloves of the South Star! The popular fist enhancers in a fun, harmless, family-friendly variant! Coming to an arcade machine near you!
/:cl:

### CODE REVIEW

**Comment by FernandoJ8** [comment]
Don't like it, close it down

File: fulp_modules/ZFulpcode/modules/granters/granters.dm [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Overwrites martial arts granters (granters.dm) to prevent South Star users from getting other martial arts
+// while they are wearing the gloves.
+
+/obj/item/book/granter/martial/on_reading_finished(mob/user)
```
Comment (JohnFulpWillard) [inline fulp_modules/ZFulpcode/modules/granters/granters.dm:4]: whats the point of this instead of just using allow_temp_override = FALSE

**Comment by SgtHunk** [comment]
Why not PR to TG?

File: fulp_modules/ZFulpcode/modules/granters/granters.dm (hunk 1 above)
Comment (JosephJomama) [reply fulp_modules/ZFulpcode/modules/granters/granters.dm:4]: I tried that, but that only overrides martial arts that are taught as temporary martial arts. Granters like those that give you CQC or Carp teach it to you as a non-temporary martial art.

**Comment by JosephJomama** [comment]
> Why not PR to TG?

I could, but I noticed that we had some exclusive Fulp toys (batong, supermatter) and figured I'd add to the collection.

**Comment by SgtHunk** [comment]
> I could, but I noticed that we had some exclusive Fulp toys (batong, supermatter) and figured I'd add to the collection. 

The batong is fulp-exclusive for a meme. But IMO the Supermatter toy should have been PR'd to TG, alongside the food items we have.

**Comment by JosephJomama** [comment]
> > I could, but I noticed that we had some exclusive Fulp toys (batong, supermatter) and figured I'd add to the collection.
> 
> The batong is fulp-exclusive for a meme. But IMO the Supermatter toy should have been PR'd to TG, alongside the food items we have.

Fair. I'll leave this up to the maintainers, I was under the impression that there was a precedent for adding new Fulp exclusive toys.

File: fulp_modules/ZFulpcode/modules/granters/granters.dm (hunk 1 above)
Comment (ATH1909) [reply fulp_modules/ZFulpcode/modules/granters/granters.dm:4]: you do realize that not all martial arts come from granters, right?

the warden's krav maga, for example, comes from their gloves, a chef's CQC comes from a skillchip, and so on.

Comment (ATH1909) [reply fulp_modules/ZFulpcode/modules/granters/granters.dm:4]: you could alternatively just put on the gloves AFTER learning sleeping carp or whatever, no?

File: fulp_modules/features/toys/code/toys.dm [hunk 2]
```diff
@@ -24,3 +24,24 @@
...
+	desc = "Just looking at these fills you with an urge to pretend to beat the shit out of people."
+	var/datum/martial_art/northstar/style = new
+
+/obj/item/clothing/gloves/rapid/toy/equipped(mob/user, slot)
+	..()
+	if(!ishuman(user))
+		return
+	if(slot == ITEM_SLOT_GLOVES)
+		var/mob/living/student = user
+		student.mind.martial_art.allow_temp_override = TRUE
```
Comment (ATH1909) [inline fulp_modules/features/toys/code/toys.dm:39]: uhhhhhhhhhh

ain't this gonna be a permanent effect, allowing you to override martial arts that aren't meant to be overridden, even after taking the gloves off?

File: fulp_modules/features/toys/code/martial/northstar.dm [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/datum/martial_art/northstar
+	name = "South Star"
+	id = "south star"
+	allow_temp_override = FALSE
+
+/datum/martial_art/northstar/harm_act(mob/living/A, mob/living/D)
```
Comment (ATH1909) [inline fulp_modules/features/toys/code/martial/northstar.dm:6]: gloves of the north star also affect your shoves while you're in combat mode

**Comment by ATH1909** [comment]
I'd recommend just scrapping the rapid punching part of this PR and instead just making gloves of the south star be boxing glove subtypes that make you say a phrase whenever you punch someone with them or something.

**Comment by Helianthus00** [comment]
Yes this would be a good idea to pr to tg first. If they reject it, I'll be more likely to merge it here, but already our fulp toys/food have caused some issues.

File: fulp_modules/features/toys/code/toys.dm (hunk 2 above)
Comment (JosephJomama) [reply fulp_modules/features/toys/code/toys.dm:39]: Yeah, good catch, after unlearning the art I need to set it back to normal

**Pull request closed without merging**